### PR TITLE
Feat: Grab comments above global symbols and show them on hover

### PR DESCRIPTION
**Pull request by WilsonZiweiWang**
This PR adds the comments above the symbols (Variables and functions) defined in the global scope in the hover definition.
**Sample results**:
Show all comments in the current file and included files
image
The comments are appended at the end
image

Caveat:
The results won't reflect right after modifications on the comments in the included files because it needs to run `analyze()` to update the values. Currently, it needs additional input to trigger it.

What I have in mind are these two:
1. Somehow trigger the `analyze()` on the current file when the included files are modified.
2. Maintain the references of the comments in the included files so that the current file always gets the updated comments.

Despite the caveat, the feature itself is ready for review.

### CODE REVIEW

**Comment by deribaucourt** [comment]
Looks very promising! The idea of this feature is extending the hover documentation functionality to variables not covered by the poky doc, mostly from community layers. The goal should not be to display all comments. We have the Go to definition feature for navigating definitions and related comments.

A few comments:

1. If a hover documentation from the Yocto docs are present, don't add comments. We already have the variable's doc from an official source.
2. You can hence remove the "Comments" title
3. Having multiple files comments is nice but a bit cumbersome. You should consider only keeping one with the logic we have discussed on Mattermost: .bbclass > .conf > .inc > .bb > .bbappend. If multiple files of the same extension provide comments, pick one at random. We have no way of determining the order yet.
4. It's alright if analyze() is not called automatically unless a specific event happens in the current file. It can be one of: the user saves the file; the user modifies the file.

**Review by deribaucourt — COMMENTED** [review]
Code looks good

**Comment by WilsonZiweiWang** [comment]
> Looks very promising! The idea of this feature is extending the hover documentation functionality to variables not covered by the poky doc, mostly from community layers. The goal should not be to display all comments. We have the Go to definition feature for navigating definitions and related comments.
> 
> A few comments:
> 
> 1. If a hover documentation from the Yocto docs are present, don't add comments. We already have the variable's doc from an official source.
> 2. You can hence remove the "Comments" title
> 3. Having multiple files comments is nice but a bit cumbersome. You should consider only keeping one with the logic we have discussed on Mattermost: .bbclass > .conf > .inc > .bb > .bbappend. If multiple files of the same extension provide comments, pick one at random. We have no way of determining the order yet.
> 4. It's alright if analyze() is not called automatically unless a specific event happens in the current file. It can be one of: the user saves the file; the user modifies the file.

1,2: Done.
3: Which `.conf` files should be checked?
4. Currently, only the modifications on the current file will reflect immediately. Others will require additional input on the current file.

**Comment by deribaucourt** [comment]
Currently we don't bring .conf files to our recipes list so ignore them for now. Could become relevant when we have recipe-specific scan with the full include list.

File: server/src/connectionHandlers/onHover.ts [hunk 1]
```diff
@@ -105,8 +107,19 @@ function getGlobalSymbolComments (uri: string, word: string): string | null {
...
+      // higher priority comments replace lower ones
+      priority.reverse().forEach((ext) => {
+        const commentsForExt = allCommentsForSymbol.filter((item) => path.parse(item.uri).ext === ext)
+        if (commentsForExt.length > 0) {
+          commentsToShow = commentsForExt
+        }
+      })
+
+      if (commentsToShow.length > 0) {
+        return `${commentsToShow.map((item) => item.comments.map(comment => comment.slice(1)).join('\n') + `\n\nSource: ${item.uri.replace('file://', '')} \`L: ${item.line + 1}\``).join('\n___\n')}`
```
Comment (deribaucourt) [inline server/src/connectionHandlers/onHover.ts:133]: This doesn't leave just 1 comment if there are multiple definitions in the same file. You should keep the first one in that case.

**Review by deribaucourt — CHANGES_REQUESTED** [review]

File: server/src/connectionHandlers/onHover.ts (hunk 1 above)
Comment (WilsonZiweiWang) [reply server/src/connectionHandlers/onHover.ts]: Done.

**Review by deribaucourt — APPROVED** [review]

**Pull request merged**